### DISCUSSION TITLE
Want to use as backup tool

### DISCUSSION THREAD

**Discussion by ajazalise** [Mimetis/Dotmim.Sync#937] [Q&A]
Hi, I want to use it as a backup tool where a client database will have access to Internet occasionally and both will be MSSQL DBs. I would like to use uploadonly sync to send changes to server database from client system. And I would like to use the server database as backup to restore in case of disaster. Wonder if it will be smooth to restore or you think its not what dotmim.sync is for. I will still be interested in using it if I can get it working with a few modifications if required to make sever db restore-able.
Thanks

**Comment by carlosxjose** [comment]
When you synchronize with no uploads is like a re-initialization and if the Snapshot option to set it up daily is on and active you don't need a backup...

When I used MSSQL replication every time we had synchronize issues just put on my apps for the user this options;

-Synchronize (normal)
-Reinicializate upload (Conflicts but still need to send data)
-Reinicializate no upload (Conflicts... Everything will be lost and it will download the most updated database snapshot)

**Comment by Mimetis** [comment]
Hey @ajazalise 

As @carlosxjose mentioned, using the snapshot option would be a good option to not pressure too much the server database when a reinitialize is called.
Or you can just sync data (in an Upload only mode) with the server, then make a classic SQL backup, then let the client download the backup and restore it, in case of any crash or whatever can happen to your client.

That's being said, if I understand correctly, your architecture is more or less a 1 to 1 sync?
I mean, you have a client database, and the server database is only serving one client database?

Using **DMS** will generate a lot of stuff, like stored procedures, triggers and so on ...
That means you will have to deal with that.
If your schema is changing (adding / removing columns / tables) will require you to `Provision` or `Deprovision` the DMS metadata to ensure **DMS** is taking care of the new / deleted columns & tables.

It's definitely possible but require you to be aware of what is done by DMS and the challenges to trade with, when schema is evolving.

If you decide to implement it, let me know, I will try to help you figure out the challenges you are facing during the implementation.

**Reply by ajazalise** [reply]
Thank you for the kind response.
Yes, architecture is 1 to 1, unidirectional and no DB alter.
I wonder if there is a defined procedure to make a synced database free from DMS related meta data and structure?
Sure I will let you know, infact I have a uploadonly code testing but have not yet done a cycle of initialize-sync-crash-restore

**Reply by Mimetis** [reply]
If you are using **Sql Change Tracking** feature from **SQL Server** on both side, there is no **Triggers** or **Tracking tables** generated as we are relying on this feature to track changes.

That being said, stored procedures are still needed in any cases, that you will need to maintain, through the **DMS** sdk, when your schema is evolving.

**Comment by Mimetis** [comment]
As this is more a question than an issue, I'm moving it to the discussions tab.
